src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3355,18 +3356,33 @@ private void AddSynthesizedMembers(MembersAndInitializersBuilder builder, Declar
                 case TypeKind.Class:
                 case TypeKind.Interface:
                 case TypeKind.Submission:
-                case TypeKind.Extension:
                     AddSynthesizedTypeMembersIfNecessary(builder, declaredMembersAndInitializers, diagnostics);
                     AddSynthesizedConstructorsIfNecessary(builder, declaredMembersAndInitializers, diagnostics);
                     break;
 
+                case TypeKind.Extension:
+                    AddSynthesizedExtensionMarkerIfPossible(builder, declaredMembersAndInitializers, diagnostics);
```

Comment:
>AddSynthesizedExtensionMarkerIfPossible [](http://example.com/codeflow?start=20&length=39)

I do not think we should be doing this. The marker should be emitted, but I don't think it should be part of the model.